activerecord/lib/active_record/locking/optimistic.rb

```diff
@@ -107,8 +107,9 @@ def _update_record(attribute_names = self.attribute_names)
             affected_rows
 
           # If something went wrong, revert the locking_column value.
-          rescue Exception
-            send(lock_col + "=", previous_lock_value.to_i)
+          rescue
```

Comment:
Why changing this? I believe we want to unset the value in any exception, not just subclasses of `StandardError`